CyclopsNuclearReactor/CyNukeChargeManager.cs

```diff
@@ -0,0 +1,150 @@
+
+        public readonly List<CyNukeReactorMono> CyNukeReactors = new List<CyNukeReactorMono>();
+
+        public CyNukeChargeManager(SubRoot cyclops)
+        {
+            Cyclops = cyclops;
+            InstanceID = cyclops.GetInstanceID();
+
+            QuickLogger.Debug($"Created new CyNukeChargeManager for Cyclops {InstanceID}");
+        }
+
+        internal void SyncReactorsExternally()
+        {
+            var _tempCache = new List<CyNukeReactorMono>();
+
+            CyNukeReactorMono[] foundReactors = Cyclops.GetAllComponentsInChildren<CyNukeReactorMono>();
+
+            foreach (CyNukeReactorMono reactor in foundReactors)
+            {
+                if (_tempCache.Contains(reactor))
+                    continue; // This is a workaround because of the object references being returned twice in this array.
+
+                _tempCache.Add(reactor);
+
+                if (reactor.ParentCyclops == null)
+                {
+                    QuickLogger.Debug("Cyclops Nuclear Reactor synced externally");
+                    // This is a workaround to get a reference to the Cyclops into the CyNukeReactorMono
+                    reactor.ConnectToCyclops(Cyclops, this);
+                }
+            }
+        }
+
+        internal static void SyncReactors()
+        {
+            foreach (CyNukeChargeManager mgr in Managers)
+                mgr.SyncReactorsExternally();
+        }
+
+        #region ICyclopsCharger Methods
+
+        public float ProducePower(float requestedPower)
+        {
+            if (CyNukeReactors.Count == 0)
+                return 0f;
+
+            float powerDeficit = requestedPower;
+            float producedPower = 0f;
+
+            foreach (CyNukeReactorMono reactor in CyNukeReactors)
+            {
+                if (!reactor.HasPower())
+                    continue;
+
+                producedPower += reactor.ProducePower(ref powerDeficit);
+            }
+
+            return producedPower;
+        }
+
+        public bool HasPowerIndicatorInfo()
+        {
+            if (CyNukeReactors.Count == 0)
+                return false;
+
+            foreach (CyNukeReactorMono reactor in CyNukeReactors)
+            {
+                if (reactor.HasPower())
+                    return true;
+            }
+
+            return false;
+        }
+
+        public Atlas.Sprite GetIndicatorSprite()
+        {
+            return indicatorSprite;
+        }
+
+        public string GetIndicatorText()
+        {
+            if (CyNukeReactors.Count == 0)
+                return string.Empty;
+
+            if (CyNukeReactors.Count == 1)
+                return CyNukeReactors[0].PowerIndicatorString();
+
+            string value = string.Empty;
+
+            for (int i = 0; i < CyNukeReactors.Count; i++)
+                value += $"{CyNukeReactors[i].PowerIndicatorString()}\n";
```

Comment:
I'm considering keeping the max number of reactors per cyclops as a fixed `2`.  
If that ends up being the case, this can be all be simplified.

Reply:
Let's leave this alone for now.